lib/HotKeysHOC.js

```diff
@@ -1 +1,79 @@
-// TODO: Expose HotKeys functionality as HoC
\ No newline at end of file
+import React, {PureComponent} from 'react';
+import HotKeys from './HotKeys';
+import _ from 'lodash';
+
+const DEFAULT_STATE = {
+  keyboardAction: null,
+  keyboardEvent: null,
+};
+
+/**
+* HotKeysHOC is an HOC that provides the wrappedComponent with the ability to implement keyboard actions
+* without the user wrapping every component with a <HotKeys> component individually
+*
+* See examples/master/HOCWrappedNode.js for an example implementation
+* Follow the steps below to use the HOC:
+*
+* 1. Declared an ACTION_KEY_MAP
+* // example action to key map for a dropdown menu
+* const HIDE_CHILD = 'hideChild';
+* const SHOW_CHILD = 'showChild';
+*
+* const ACTION_KEY_MAP = {
+*     [HIDE_CHILD]: 'esc',
+*     [SHOW_CHILD]: ['down', 'space', 'enter']
+* };
+*
+* 2) Add or amend `componentWillReceiveProps({keyboardAction, keyboardEvent})`
+*
+* Declare let variable `keyboardActionHelper` with keys of the ACTION_KEY_MAP
+* Object keys for keyboardActionHelper === Object keys for ACTION_KEY_MAP
+* Values for keyboardActionHelper are handler methods that the user invokes upon keyboardAction
+*
+* Add the code snippet below to the end of `componentWillReceiveProps`:
+* if (_.has(keyboardActionHelper, keyboardAction)) {
+*     keyboardActionHelper[keyboardAction]();
+* }
+*
+* 3) Wrap the Component with withKeyboardSupport
+* - Remove `export default` from Component class definition'
+* - Insert `export default withKeyboardSupport(ACTION_KEY_MAP)(<insert_Component_name>);` at the end of `Component.js`
+* - Alternatively, use `const WrappedApp = HotKeysHOC(ACTION_KEY_MAP)(<componentName>);` to wrap a component
+*
+* @summary An HOC that provides the wrappedComponent with the ability to implement keyboard actions
+*/
+const HotKeysHOC = (keyMap) => ((Component) =>
```

Comment:
I am not a particular fan of `HotKeysHOC` as a name for the higher order component. I've always used the convention of camelCase names for HOC to make it easier to conceptualise them as functions. This [seems to be consistent with the React docs](https://reactjs.org/docs/higher-order-components.html)

I also don't think it's necessary to have "HOC" in the name.

I suggest something like "withHotKeys" to remain consistent with the package and component name, and apparent community HOC naming conventions - unless you have a strong case against?

Reply:
the internal name we're using for Eventbrite is `withKeyboardSupport`, but `withHotkeys` might be more consistent with the library name. I can change it to `withHotkeys`

Reply:
Can we keep the library name capitalisation the same as well (`withHotKeys`)?